src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/common.kt

```diff
@@ -465,35 +465,29 @@ fun refundTransaction(
 
   return Mono.just(tx)
     .flatMap { transaction ->
-      val authorizationRequestId = transactionAuthorizationRequestData.authorizationRequestId
       when (transactionAuthorizationRequestData.paymentGateway) {
         TransactionAuthorizationRequestData.PaymentGateway.NPG ->
-          appendNpgRefundRequestedEventIfNeeded(
-              transaction, transactionsEventStoreRepository, transactionsViewRepository, npgService)
-            .flatMap { (tx, _) -> refundTransactionNPG(tx, refundService) }
-        TransactionAuthorizationRequestData.PaymentGateway.REDIRECT -> {
-          appendRefundRequestedEventIfNeeded(
-              tx, transactionsEventStoreRepository, transactionsViewRepository)
-            .flatMap {
-              refundService.requestRedirectRefund(
-                transactionId = transaction.transactionId,
-                touchpoint =
-                  transaction.clientId.effectiveClient.let {
-                    when (it) {
-                      Transaction.ClientId.CHECKOUT_CART -> Transaction.ClientId.CHECKOUT.name
-                      Transaction.ClientId.CHECKOUT,
-                      Transaction.ClientId.IO -> it.name
-                      else ->
-                        throw IllegalArgumentException(
-                          "Cannot determine touch point: [$it] for redirect transaction refund")
-                    }
-                  },
-                pspTransactionId = transactionAuthorizationRequestData.authorizationRequestId,
-                paymentTypeCode = transactionAuthorizationRequestData.paymentTypeCode,
-                pspId = transactionAuthorizationRequestData.pspId)
-            }
+          refundTransactionNPG(
```

Comment:
It seems that the call to `appendNpgRefundRequestedEventIfNeeded` is gone, is it ok?

Reply:
yes, the refund transaction code is called by refund requested/refund error event handlers. Those handlers already require a transaction to be in refund requested / refund error event so this code will never write this event.